scripts/battlefields/Apollyon/cs_apollyon.lua

```diff
@@ -0,0 +1,312 @@
+        end
+    end
+end
+
+function content:onBattlefieldTick(battlefield, tick)
+    Battlefield.onBattlefieldTick(self, battlefield, tick)
+
+    -- Do nothing unless the battlefield has been locked due to engaged
+    if battlefield:getStatus() ~= xi.battlefield.status.LOCKED then
+        return
+    end
+
+    -- Do nothing if aggroTime hasn't been set or aggroTime hasn't been reached yet
+    local now = os.time()
+    local aggroTime = battlefield:getLocalVar("AutoAggroTime")
+    if aggroTime == 0 or aggroTime > now then
+        return
+    end
+
+    -- Get which player is going to be aggrod
+    local player = GetPlayerByID(battlefield:getLocalVar("AutoAggroTarget"))
+    if player:isDead() then
+        -- Need to find a new target
+        local players = battlefield:getPlayers()
+        for _, battlefieldPlayer in pairs(players) do
+            if battlefieldPlayer:isAlive() then
+                player = battlefieldPlayer
+                break
+            end
+        end
+    end
+
+    -- Determine which boss should aggro next
+    local previousBoss = battlefield:getLocalVar("AutoAggro")
+    local nextBoss = 0
+    if previousBoss == ID.CS_APOLLYON.mob.CARNAGECHIEF_JACKBODOKK then
+        nextBoss = ID.CS_APOLLYON.mob.DEE_WAPA_THE_DESOLATOR
+    elseif previousBoss == ID.CS_APOLLYON.mob.DEE_WAPA_THE_DESOLATOR then
+        nextBoss = ID.CS_APOLLYON.mob.NAQBA_CHIRURGEON
+    else
+        nextBoss = ID.CS_APOLLYON.mob.CARNAGECHIEF_JACKBODOKK
+    end
+    local boss = GetMobByID(nextBoss)
+
+    battlefield:setLocalVar("AutoAggro", boss:getID())
+    battlefield:setLocalVar("AutoAggroTime", os.time() + utils.minutes(7))
+    battlefield:setLocalVar("AutoAggroTarget", player:getID())
+
+    if boss:isAlive() then
+        boss:updateEnmity(player)
+    end
+end
+
+function content:onBattlefieldWipe(battlefield, players)
+    Battlefield.onBattlefieldWipe(self, battlefield, players)
+    battlefield:setLocalVar("AutoAggro", 0)
+    battlefield:setLocalVar("AutoAggroTime", 0)
+    battlefield:setLocalVar("AutoAggroTarget", 0)
+end
+
+local supportSetup = function(supportOffsets, otherSupportOffsets, battlefield, mobs)
+    local boss = mobs[1]
+    local bossID = boss:getID()
+
+    -- Watch for support mobs to be dead and once they are use "Lateral Slash" to summon the next support group
+    boss:addListener("COMBAT_TICK", "COMBAT_TICK_SUPPORT", function(mobArg)
+        if boss:isDead() then
+            return
+        end
+
+        local now = os.time()
+        if now < boss:getLocalVar("supportTimer") then
+            return
+        end
+
+        local group = boss:getLocalVar("supportGroup")
+        local offsets = group == 0 and supportOffsets or otherSupportOffsets
+        for _, offset in ipairs(offsets) do
+            if GetMobByID(bossID + offset):getStatus() ~= xi.status.DISAPPEAR then
+                return
+            end
+        end
+
+        boss:useMobAbility(603)
```

Comment:
Surely, they don't actually use lateral slash which is a tonberry TP move

Consider using `boss:injectActionPacket` to send the action packet to give the client what it needs and a timer to replace the weaponskill listener

If injectActionPacket doesn't do what we need then we can add functionality to do it

Reply:
I didn't know about `injectActionPacket` I'll give that a try. Here's what the `0x028 Action By NPC` packet looks like.
![image](https://user-images.githubusercontent.com/1189557/195944358-7dc9572d-e05e-499a-8558-5c80d34a8f83.png)